src/client/localplayer.cpp

```diff
@@ -316,8 +316,16 @@ void LocalPlayer::move(f32 dtime, Environment *env,
 	if (!(is_valid_position && is_valid_position2)) {
 		is_climbing = false;
 	} else {
-		is_climbing = (nodemgr->get(node.getContent()).climbable ||
-			nodemgr->get(node2.getContent()).climbable) && !free_move;
+		bool climbable_upper = nodemgr->get(node.getContent()).climbable;
+		bool climbable_lower = nodemgr->get(node2.getContent()).climbable;
+		is_climbing = (climbable_upper || climbable_lower) && !free_move;
+		if (is_climbing) {
+			if (climbable_lower) {
+				node_climb_factor = nodemgr->get(node2.getContent()).climb_factor;
+			} else {
+				node_climb_factor = nodemgr->get(node.getContent()).climb_factor;
+			}
+		}
```

Comment:
Slightly shorter code, where `node` and `node2` are only used once, hence reduces potential confusions about which one's the upper or lower node.
```suggestion
		const ContentFeatures &cf_upper = nodemgr->get(node.getContent());
		const ContentFeatures &cf_lower = nodemgr->get(node2.getContent());
		is_climbing = (cf_upper.climbable || cf_lower.climbable) && !free_move;
		if (is_climbing) {
			node_climb_factor = cf_lower.climbable
				? cf_lower.climb_factor : cf_upper.climb_factor;
		}
```